src/Realm.C

```diff
@@ -921,26 +921,15 @@ void
 Realm::setup_nodal_fields()
 {
 #ifdef NALU_USES_HYPRE
-  hypreGlobalId_ = &(meta_data().declare_field<HypreIDFieldType>(
-    stk::topology::NODE_RANK, "hypre_global_id"));
+  fieldManager_->register_field("hypre_global_id", meta_data().get_parts());
 #endif
-  tpetGlobalId_ = &(meta_data().declare_field<TpetIDFieldType>(
-    stk::topology::NODE_RANK, "tpet_global_id"));
+  fieldManager_->register_field("tpet_global_id", meta_data().get_parts());
+  fieldManager_->register_field("nalu_global_id", meta_data().get_parts());
 
-  // register global id and rank fields on all parts
-  const stk::mesh::PartVector parts = meta_data().get_parts();
-  for (size_t ipart = 0; ipart < parts.size(); ++ipart) {
-    naluGlobalId_ = &(meta_data().declare_field<GlobalIdFieldType>(
-      stk::topology::NODE_RANK, "nalu_global_id"));
-    stk::mesh::put_field_on_mesh(*naluGlobalId_, *parts[ipart], nullptr);
-
-#ifdef NALU_USES_HYPRE
-    stk::mesh::put_field_on_mesh(*hypreGlobalId_, *parts[ipart], nullptr);
-#endif
-    stk::mesh::put_field_on_mesh(*tpetGlobalId_, *parts[ipart], nullptr);
-    stk::mesh::field_fill(
-      std::numeric_limits<LinSys::GlobalOrdinal>::max(), *tpetGlobalId_);
-  }
+  auto globId = fieldManager_->get_field_ptr("nalu_global_id");
+  stk::mesh::field_fill(
+    std::numeric_limits<LinSys::GlobalOrdinal>::max(),
```

Comment:
Another pre-existing thing... 0 would be a better initial value than max (assuming this initial value should be obviously invalid), because stk-mesh knows that 0 is an invalid ID (and will throw in certain places if it encounters 0) whereas it doesn't know that max is invalid.